cli/azd/pkg/apphost/generate_types.go

```diff
@@ -102,10 +103,15 @@ type genBicepTemplateContext struct {
 	CosmosDbAccounts                map[string]genCosmosAccount
 }
 
+type genContainerAppEnvVar struct {
```

Comment:
Instead of doing this, it feels like it might be cleaner to just add a `.Secrets` property to the template context, which is an `map[string]string`, This would give us both `.Env` (the non secret env-vars) and `.Secrets` the secrets.  We can loop over the `.Secrets` map to build out the `secrets:` section of the ACA yaml, and then loop over both `.Env` and `.Secrets` when building the `env` block to either add the literal values (for `.Env`) or references to the secrets (for `.Secrets`).

Would that work?

I think it would let us get rid of the `hasSecrets` method, as well, since that would now just be a len check on .Secrets.

Reply:
Updated to use a map for secrets